flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/ReconciliationUtils.java

```diff
@@ -281,6 +284,17 @@ public static boolean isUpgradeModeChangedToLastStateAndHADisabledPreviously(
                 && !HighAvailabilityMode.isHighAvailabilityModeActivated(observeConfig);
     }
 
+    public static <SPEC extends AbstractFlinkSpec> SPEC getLastSpec(
+            AbstractFlinkResource<SPEC, ?> deployment) {
+        var reconciliationStatus = deployment.getStatus().getReconciliationStatus();
+        var reconciliationState = reconciliationStatus.getState();
+        if (reconciliationState != ReconciliationState.ROLLED_BACK) {
+            return reconciliationStatus.deserializeLastReconciledSpec();
+        } else {
+            return reconciliationStatus.deserializeLastRollbackSpec();
```

Comment:
We should use `lastStableSpec` and avoid introducing a new status field. The lastStableSpec is meant exactly for rollback purposes.

Reply:
I rely on the `lastStableSpec` to do the rollback.
It is indeed the spec used to roll back to a working state.
But once the job is rolled back I need to be able to detect changes to the current spec (the one which has lead to the rollback). As the job spec has not been updated we need to be able to compare it with a specific stored spec from which we will be able to do comparison and find changes
I can't rely on the `lastReconcileSpec` or `lastStableSpec` as they are not align with the job spec.
This is why I introduce this `lastRollbackSpec` which contains the job spec leading to failure.

Reply:
You should always compare to the `lastReconciledSpec` I think that was the last spec that was reconciled and submitted by the user.

Reply:
As long as you do not update the `lastReconciledSpec` during the rollback the logic should be good

Reply:
The lastReconcileSpec is updated now as we rely on same code path than the upgrade one.
I will take a look to not update it but it looks quite strange to me to finally not have the lastReconcile being aligned with the "current" running spec

Reply:
`lastReconciled` spec is not `current` spec. Current spec is already in the spec, lastReconciled is what was last reconciled by the operator based on what the user provided. Its sole purpose is to be able to do the above mentioned comparison. We should try not to change the semantics if possible.